src/main/java/edu/kit/compiler/parser/Parser.java

```diff
@@ -123,160 +193,201 @@ private void parseBlock() {
                 check(TokenType.Identifier, TokenType.Identifier) ||
                 check(TokenType.Identifier, TokenType.Operator_BracketL, TokenType.Operator_BracketR)) {
                 // LocalVariableDeclarationStatement
-                parseType();
-                expect(TokenType.Identifier);
+                Token firstToken = tokenStream.get();
+                DataType variableType = parseType();
+                Optional<ExpressionNode> expr = Optional.empty();
+                Token name = expect(TokenType.Identifier);
                 if (tokenStream.get().getType() == TokenType.Operator_Equal) {
                     tokenStream.next();
-                    parseExpression();
+                    expr = Optional.of(parseExpression());
                 }
                 expect(TokenType.Operator_Semicolon);
+                stmts.add(new LocalVariableDeclarationStatementNode(
+                    firstToken.getLine(), firstToken.getColumn(), variableType, name.getIntValue().get(), expr, false
+                ));
             } else {
                 // Statement
-                parseStatement();
+                Optional<StatementNode> stmt = parseStatement();
+                stmt.ifPresent(node -> stmts.add(node));
```

Comment:
How fast are lambdas in Java? Seems like it would be simple to write it without here.